src/main/java/io/github/opencubicchunks/cubicchunks/mixin/core/common/world/MixinPathNavigationRegion.java

```diff
@@ -0,0 +1,152 @@
+package io.github.opencubicchunks.cubicchunks.mixin.core.common.world;
+
+import io.github.opencubicchunks.cubicchunks.chunk.ICubeProvider;
+import io.github.opencubicchunks.cubicchunks.chunk.cube.EmptyCube;
+import io.github.opencubicchunks.cubicchunks.chunk.util.CubePos;
+import io.github.opencubicchunks.cubicchunks.server.CubicLevelHeightAccessor;
+import io.github.opencubicchunks.cubicchunks.utils.Coords;
+import io.github.opencubicchunks.cubicchunks.world.CubeCollisionGetter;
+import net.minecraft.core.BlockPos;
+import net.minecraft.core.SectionPos;
+import net.minecraft.util.Mth;
+import net.minecraft.world.level.BlockGetter;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.level.PathNavigationRegion;
+import net.minecraft.world.level.chunk.ChunkAccess;
+import org.jetbrains.annotations.Nullable;
+import org.spongepowered.asm.mixin.Final;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Mutable;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.Redirect;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+
+@Mixin(PathNavigationRegion.class)
+public abstract class MixinPathNavigationRegion implements CubeCollisionGetter {
+
+    @Mutable @Shadow @Final protected int centerX;
+    @Mutable @Shadow @Final protected int centerZ;
+    @Shadow protected boolean allEmpty;
+    @Shadow @Final protected Level level;
+    private int centerY;
+    private int diameter;
+    private boolean isCubic;
+    private int minY;
+    private int height;
+
+    private ChunkAccess[] cubes;
+
+//    @Redirect(method = "<init>", at = @At(value = "NEW", target = "net/minecraft/world/level/chunk/ChunkAccess"))
+//    private ChunkAccess[][] no2DArray(Level level, BlockPos blockPos, BlockPos blockPos2) {
+//        if (!((CubicLevelHeightAccessor) level).isCubic()) {
+//            return new ChunkAccess[][]
+//        }
+//    }
```

Comment:
Can this be removed?

Reply:
It'll need to be added and made to return null if we're cubic.